ddtrace/tracer/option.go

```diff
@@ -107,6 +113,10 @@ func defaults(c *config) {
 	}
 	if v := os.Getenv("DD_SERVICE"); v != "" {
 		c.serviceName = v
+		c.envService = v
+		if ver := os.Getenv("DD_VERSION"); ver != "" {
```

Comment:
Why can't this be set without `DD_SERVICE`? A user might set his service programmatically using `WithServiceName`, but set his version differently using env. vars.

Reply:
This is just a requirement I was given for this task, version should only be set if the span's service is `DD_SERVICE`. I'm not 100% sure on the reasoning behind it, but I suspect they only want versions attached to spans created directly by the application and not integration spans which set their own service name.
You can see they intend to do the same thing in Python: https://github.com/DataDog/dd-trace-py/pull/1222/files#diff-ba5f6c17564de9d9f60424d8ddda624bR390

Reply:
Correct. A service is a service. `DD_SERVICE` or otherwise. They are not two separate things.

> One thing I'm wondering is if we want to provide an API for WithServiceVersion. That way folks can set version programmatically, not just through the env var. Basically it would be equivalent to the user providing DD_VERSION, and it would take priority because it's a "manual setting".

Yep, we probably should.

Reply:
Thanks, @jdgumz and @gbbr. This clears things up and makes the implementation cleaner too. I'll get this fixed up and add the `WithServiceVersion` `StartOption.`